network-mux/src/Network/Mux/Bearer/AttenuatedChannel.hs

```diff
@@ -92,6 +89,8 @@ writeQueueChannel QueueChannel { qcWrite } msg =
         Nothing -> return False
         Just q  -> writeTQueue q msg
                 >> case msg of
+                      -- Match SO_LINGER set with 0 interval
+                      -- and close both ends without any handshake
```

Comment:
I'd say in a way we are already simulating this case by writing the message and not waiting on receiving some sort of Ack. Of course since this is a simulation environment we do not actually need to do much because the message doesn't go through the wire.

Reply:
Addressed!

Reply:
> I'd say in a way we are already simulating this case by writing the message and not waiting on receiving some sort of Ack.

That's not right; resting a connection does not make it wait on an ack, the kernel will just forget the state of the connection when it sends the packet with `RST` option set.   If the receiver does not get the packet it will get an error when next time it will try to send a packet (a reset will be send as a reply).

Reply:
I will experiment with simulating a lost message like you suggested and see if that still passes the test. The problem before was that we were waiting for an Ack and that could take very long. Hopefully just closing and losing the MsgClose message will not delay it much

Reply:
Tried it and the tests are passing 😄 I think now it simulates the LINGER option more faithfully.